siriuspy/siriuspy/devices/bpm_eq.py

```diff
@@ -0,0 +1,581 @@
+        ax.legend(loc='best', ncol=2, fontsize='x-small')
+        ax.set_title(f'Acq Strategy: ' + self.AcqStrategies._fields[acqs])
+        fig.tight_layout()
+        return fig, ax
+
+    def plot_antennas_mean(self):
+        """."""
+        posx_gain = self.data.get('posx_gain')
+        posy_gain = self.data.get('posx_gain')
+        gacq = self.data.get('gains_acq')
+        gini = self.data.get('gains_init')
+        gnew = self.data.get('gains_new')
+        antm = self.data.get('antennas_mean')
+        if gnew is None:
+            self._log('ERR:Must acquire data and process first.')
+            return None, None
+
+        gains = (1, gacq, gini, gnew)
+        titles = ('Unit Gain', 'Acquisition Gain', 'Old Gain', 'New Gain')
+        labs = list('ABCDXYS')
+        labs[-1] = 'Sum'
+        labs = [lab + ' [a.u.]' for lab in labs]
+        labs[4] = r'X [$\mu$m]'
+        labs[5] = r'Y [$\mu$m]'
+
+        antm /= 10**(_np.floor(_np.log10(antm.max())))
+
+        fig, axs = _mplt.subplots(
+            len(labs), len(gains), figsize=(16, 10), sharex=True, sharey='row')
+        for j, gain in enumerate(gains):
+            val = antm * gain
+            for i in range(4):
+                ax = axs[i, j]
+                ld = ax.plot(val[:, i, 0], 'o-')[0]
+                li = ax.plot(val[:, i, 1], 'o-')[0]
+                if not i and not j:
+                    ld.set_label('Direct')
+                    li.set_label('Inverse')
+            posx, posy = self._estimate_orbit(antm, gain, posx_gain, posy_gain)
+            sum_ = val.sum(axis=1)
+            axs[4, j].plot(posx[:, 0], 'o-')
+            axs[4, j].plot(posx[:, 1], 'o-')
+            axs[5, j].plot(posy[:, 0], 'o-')
+            axs[5, j].plot(posy[:, 1], 'o-')
+            axs[6, j].plot(sum_[:, 0], 'o-')
+            axs[6, j].plot(sum_[:, 1], 'o-')
+            for i, lab in enumerate(labs):
+                ax = axs[i, j]
+                if not j:
+                    ax.set_ylabel(lab)
+                ax.grid(True, alpha=0.5, ls='--', lw=1)
+            axs[0, j].set_title(titles[j])
+            axs[-1, j].set_xlabel('BPM Index')
+
+        axs[0, 0].legend(loc='best', fontsize='x-small', ncol=2)
+        fig.tight_layout()
+        return fig, axs
+
+    # ------- auxiliary methods ----------
+
+    def _estimate_orbit(self, mean, gains, posx_gain, posy_gain):
+        ant = mean * gains
+        # Get pairs of antennas
+        ac = ant[:, ::2]
+        bd = ant[:, 1::2]
+        # Calculate difference over sum for each pair
+        dovs_ac = _np.diff(ac, axis=1).squeeze() / ac.sum(axis=1)
+        dovs_bd = _np.diff(bd, axis=1).squeeze() / bd.sum(axis=1)
+        # Get the positions:
+        posx = (dovs_ac - dovs_bd)
+        posy = (dovs_ac + dovs_bd)
+        # Apply Position gains and factor of two missing in previous step
+        posx *= posx_gain[:, None] / 2 / 1e3
+        posy *= posy_gain[:, None] / 2 / 1e3
+        return posx, posy
```

Comment:
Since you are saving this orbit in the data, not just using it to calculate the difference, shouldn't it also use the BPM offsets?

Reply:
right, good catch! Done!